Tyespec migration for  Commerce

welovej opened the pull request:
This PR migrates your latest version (identified by the tag in your readme.md) of swagger to TypeSpec. We already tried our best to make sure the TypeSpec represents same as previous swagger. Since we lack the business knowledge, please validate this PR again to make sure it's functional equivalent as before. The local validation step is at [Getting started | TypeSpec Azure](https://azure.github.io/typespec-azure/docs/migrate-swagger/01-get-started/#review-and-adjust-the-typespec)

Besides, TypeSpec encourages to follow [ARM guidelines](https://aka.ms/rpguidelines). Therefore, some representations in your previous swagger will be fixed to follow these guidelines. When you see differences in your local validation, please keep this note in mind.

Please reach out to [TypeSpec Discussions Channel](https://teams.microsoft.com/l/channel/19%3a906c1efbbec54dc8949ac736633e6bdf%40thread.skype/TypeSpec%2520Discussion%2520%25F0%259F%2590%25AE?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) if there is any help needed.

welovej committed:
converted

welovej committed:
could compile

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
Update routes.tsp

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

v-jiaodi committed:
Merge branch 'main' into TspMig-commerce

qiaozha reviewed (dismissed):
Please follow the new folder structure guideline and work with @v-xuto

ChenxiJiang333 committed:
Update routes.tsp

ChenxiJiang333 committed:
Create client.tsp

welovej committed:
delete before relocate

welovej committed:
relocate to conform v2 structure

v-huizhu2 committed:
feat(commerce): add Java client customizations and update emitter options

- Add @clientName decorator for UsageManager in Java
- Add @alternateType mapping for UsageSample infoFields to ClientCustomizations.InfoField
- Configure tspconfig.yaml with Azure-specific options:
  - Enable object usage for unknown types
  - Enable client-side validations
  - Set UUID as number instead of string
  - Set float32 as double instead of float

welovej committed:
Merge pull request #13 from v-huizhu2/java_mgmt_fix_commerce

java mgmt, typespec fixed for commerce

ChenxiJiang333 committed:
Update routes.tsp

ChenxiJiang333 committed:
Update client.tsp

ChenxiJiang333 commented on specification/commerce/resource-manager/Microsoft.Commerce/Commerce/routes.tsp:
```diff
@@ -29,37 +61,7 @@ interface UsageAggregatesOperationGroup {
-      @query("continuationToken")
-      continuationToken?: string;
-    },
+    Parameters = UsageAggregatesListParameters,
```
recovered since it was a fake breaking change for python

v-huizhu2 committed:
Remove Java client and alternate type from client.tsp

Removed Java client name and alternate type definition.

v-huizhu2 committed:
Add stream-style-serialization option to tspconfig

welovej committed:
Update client.tsp

welovej committed:
update

welovej committed:
Revert "update"

This reverts commit 4a1289adf5d0a3d8894167da3cc8be4f43283c85.

welovej committed:
update

welovej commented on specification/commerce/resource-manager/Microsoft.Commerce/Commerce/examples/2015-06-01-preview/GetUsageAggregatesList.json:
```diff
@@ -0,0 +1,45 @@
+      "body": {
+        "error": {
+          "code": "aaaaaaaaaaaaaaaaaaaaaa",
+          "message": "aaaaaaaaaaaaaaaaaaaa"
```
Fixed.  Please check.

v-huizhu2 commented on specification/commerce/resource-manager/Microsoft.Commerce/Commerce/tspconfig.yaml:
```diff
@@ -0,0 +1,51 @@
+    client-side-validations: true
+    uuid-as-string: false
+    float32-as-double: false
+    stream-style-serialization: false
```
client-side-validations: true – enables input validation within the generated Java client (e.g., null/length checks) so mistakes are caught before the request is sent.
uuid-as-string: false – keeps UUID-typed properties as java.util.UUID instead of demoting them to raw strings, preserving strong typing. Fix like this.
<img width="757" height="105" alt="image" src="https://github.com/user-attachments/assets/26a6906c-ddb3-4a22-80ec-90ec2608d064" />
float32-as-double: false – keeps 32-bit floating-point schema fields as float in Java rather than widening them to double, matching the service contract more closely. Fix like this.
<img width="1341" height="229" alt="image" src="https://github.com/user-attachments/assets/998bca3b-6fa6-41e4-824c-aac501d94fc0" />
stream-style-serialization: false – disables stream-based JSON serialization; the emitter will materialize payloads in memory (more compatible with most ARM services unless streaming is required). 
If these parameters are not needed, I can delete them.

v-huizhu2 replied:
<img width="1657" height="99" alt="image" src="https://github.com/user-attachments/assets/755b89bb-222f-48ad-9aa0-514115a54812" />
“stream-style-serialization: false” This parameter seems to fix this.

weidongxu-microsoft commented on specification/commerce/resource-manager/Microsoft.Commerce/Commerce/tspconfig.yaml:
```diff
@@ -0,0 +1,51 @@
+    client-side-validations: true
+    uuid-as-string: false
+    float32-as-double: false
+    stream-style-serialization: false
```
```suggestion

```

weidongxu-microsoft committed:
Update specification/commerce/resource-manager/Microsoft.Commerce/Commerce/tspconfig.yaml

jliusan committed:
add go config

jliusan committed:
revert go config

v-jiaodi committed:
Update tspconfig.yaml by removing unused properties

Removed service-dir and is-modular-library properties.

v-jiaodi committed:
update client name

v-jiaodi committed:
Enable compatibility-lro in tspconfig.yaml

v-jiaodi committed:
Remove compatibility-lro from tspconfig.yaml

Removed compatibility-lro setting from TypeSpec configuration.

welovej committed:
update

pshao25 committed:
Merge branch 'main' into TspMig-commerce
